From 77c24eb3e81f078c89f782f5e99fb6b20529a67e Mon Sep 17 00:00:00 2001
From: "Brian C. Lane" <bcl@redhat.com>
Date: Tue, 9 Sep 2014 10:53:02 -0700
Subject: [PATCH] tests: skip loop-partitioning tests when ext_range is < 2

* tests/t-lib.sh (require_partitionable_loop_device_): New function.
* tests/t8000-loop.sh: Use it.

Ported from upstream commit ff479df0cbf03

Resolves: rhbz#1139435
---
 tests/t-lib.sh      | 7 +++++++
 tests/t8000-loop.sh | 2 ++
 2 files changed, 9 insertions(+)

diff --git a/tests/t-lib.sh b/tests/t-lib.sh
index 83efdf9..dfa5ef7 100644
--- a/tests/t-lib.sh
+++ b/tests/t-lib.sh
@@ -136,6 +136,13 @@ require_built_()
   test $skip_ = yes && skip_test_ "required program(s) not built"
 }
 
+require_partitionable_loop_device_()
+{
+  case $(cat /sys/devices/virtual/block/$(basename $1)/ext_range) in
+    0|1) skip_test_ "your system does not support loop partitioning";;
+  esac
+}
+
 uid_is_privileged_()
 {
   # Make sure id -u succeeds.
diff --git a/tests/t8000-loop.sh b/tests/t8000-loop.sh
index 9dbe512..a6bfe20 100755
--- a/tests/t8000-loop.sh
+++ b/tests/t8000-loop.sh
@@ -37,6 +37,8 @@ cleanup_()
 f1=$(pwd)/1; d1=$(loop_setup_ "$f1") \
   || skip_test_ "is this partition mounted with 'nodev'?"
 
+require_partitionable_loop_device_ $d1
+
 fail=0
 
 # Expect this to succeed.
-- 
1.9.3

